api/src/resources/source.rs

```diff
@@ -123,6 +124,9 @@ pub struct NewSource<'request> {
 
     #[serde(skip_serializing_if = "Option::is_none")]
     pub sensitive_properties: Option<Vec<&'request str>>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub _type: Option<&'request str>,
```

Comment:
nit. While, I agree with this being called `_type` in the API, I am not so sure whether this should extend to client / cli as 1. this has no notion of v1. vs v2., the source here is just an abstract resource and more pragmatically, `_field` in Rust by convention is an unused field (i.e. it would not warn if you don't use it, e.g. as a tag or for FFI). `type` is a keyword though, my nit suggestion would be `type_`